db/silkworm/etl/buffer.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "buffer.hpp"
+
+namespace silkworm::etl{
+bool compareEntries(const Entry lhs, const Entry rhs) {
+    return lhs.key.compare(rhs.key) > 0;
+}
+
+void Buffer::put(ByteView key, ByteView value) {
+    size_ += value.size() + key.size();
+    entries_.push_back({key, value});
+}
+
+void Buffer::sort() {
+    std::sort(entries_.begin(), entries_.end(), compareEntries);
+    std::reverse(entries_.begin(), entries_.end());
```

Comment:
This adds a bit of oeverhead as the container is firtsly sorted ascending and then reversed.
You can easily transform this in a single `sort` pass by modding the compareEntries above to return the **inverse** of the key.comparator result